core/core.js

```diff
@@ -0,0 +1,164 @@
+    contents_: {value: undefined, // wil be vetted into existance
+		writable: true, enumerable: true, configurable: true},
+  }));
+
+  make($.physical, 'contents', function() {
+    // Return the contents of this object.  This is for VR purposes,
+    // and descendents of $.physical can override it to 'hide' certain
+    // objects from their contents.
+    $.physical_vet(this);
+    return this.contents_;
+  });
+       
+  make($.physical, 'moveto', function(dest) {
+    // Move this physical object to dest
+    //
+    // This is based loosely on moo.ca's #3:moveto and #102:bf_move
+    // Vet obj, src and dest (also does type check):
+    $.physical_vet(this);
+    var src = this.location;
+    src === null || $.physical_vet(src);
+    dest === null || $.physical_vet(dest);
+
+    // FIXME: permission checks go here.
+    // Do moveable and accept checks:
+    if (!this.moveable(dest)) {
+      var destname = (dest === null ? 'null' : dest.name);
+      throw Error(this.name + ' declined to move to ' + destname);
+    }
+    if (dest !== null && !dest.accept(this)) {
+      throw Error(dest.name + ' refused ' + this.name);
+    }
+
+    // Check proposed containment hierarchy for cycles.  We do this
+    // after the movable and accept checks because we don't want those
+    // checks in case they muck arounnd with the containment
+    // hierarchy.
+    for(var loc = dest; loc !== null; loc = loc.location) {
+      if (loc === this) {
+	throw Error("Can't put " + this.name + " inside itself.");
+      }
+    }
+
+    // Do move:
+    var i;
+    if ($.physical.isPrototypeOf(src)) {
+      while ((i = src.contents_.indexOf(this)) >= 0) {
+	src.contents_.splice(i, 1);
+      }
+    }
+    this.location = dest;
+    dest === null || dest.contents_.push(this);
+
+    // FIXME: call exitfunc
+    // FIXME: call enterfunc
+  });
+  
+  make($.physical, 'moveable', function(obj) {
+    // Returns true iff this is willing to accept obj into its contents.
+    return false;
+  });
+
+  make($.physical, 'accept', function(obj) {
+    // Returns true iff this is willing to accept obj into its
+    // contents.  This function should only be called by
+    // $.physical.moveto(), and might cause some action to occur as a
+    // result.  Other callers interested in testing to see if a move will
+    // be accepted should call .acceptable() instead.
+    return this.acceptable(obj);
+  });
+
+  make($.physical, 'acceptable', function(obj) {
+    // Returns true iff this is willing to accept obj into its
+    // contents.  By default .accept() delegates this decision to this
+    // function.  This function (and any overrides) MUST NOT cause any
+    // observable action to occur.
+    return false;
+  });
+
+  make($.physical, 'contains', function(obj) {
+    // Returns true iff obj is located in this.
+    $.physical_vet(this);
+    $.physical_vet(obj);
+    for (var loc = obj.location; loc !== null; loc = loc.location) {
```

Comment:
As in the above cases, I think the `=== null` check makes it clear that we are looking for objects whose location is 'nowhere', rather than merely a somehow invalid location.